app/models/transaction.rb

```diff
@@ -1,24 +1,35 @@
 class Transaction < ActiveRecord::Base
-  belongs_to :buyer, class_name: :Account, foreign_key: :id
-  belongs_to :receiver, class_name: :Account, foreign_key: :id2
+  belongs_to :buyer, class_name: :Account
+  belongs_to :receiver, class_name: :Account
   belongs_to :administrator, class_name: :Account, foreign_key: :admin
   self.per_page = 50
 
-  def self.log(account, bank, amount, comment: nil, admin: nil, time: Time.current)
-    batch_log([[account, amount]], bank, comment: comment, admin: admin, time: time)
+  def self.log(account, bank, amount, comment: nil, admin: nil, time: Time.current, is_tobacco: false)
+    batch_log([[account, amount]], bank, comment: comment, admin: admin, time: time, is_tobacco: is_tobacco)
   end
 
-  def self.batch_log(amounts, bank, comment: nil, admin: nil, time: Time.current) # amounts is an array [ [account, amount] ]
+  def self.batch_log(amounts, bank, comment: nil, admin: nil, time: Time.current, is_tobacco: false) # amounts is an array [ [account, amount] ]
     comment ||= ''
     transaction do
       amounts.each do |account, amount|
-        amount = (amount * 100).round(1).ceil # Screwing the customers
-        account.balance = account.balance - amount
-        account.turnover = account.turnover + amount if amount > 0
-        bank.balance = bank.balance + amount
-        bank.turnover = bank.turnover - amount if amount < 0
-        create(id: account.id, id2: bank.id, price: -amount, comment: comment, admin: admin.try(:id), date: time)
-        create(id: bank.id, id2: account.id, price: amount, comment: comment, admin: admin.try(:id), date: time)
+        buyer = amount > 0 ? account : bank
+        receiver = amount > 0 ? bank : account
+        amount = -amount if amount < 0
```

Comment:
value = amount.abs
I am a bit reluctant to use the same variable name the describe something different